docs/zkapps/tutorials/11-advanced-account-updates.mdx

```diff
@@ -0,0 +1,169 @@
+    balanceChange.assertEquals(Int64.from(0));
+  }
+```
+
+This checks that the deployUpdate is a single account update, with no children - and that it isn't creating any additional tokens, and is starting out with a balance of zero.
+
+Next, here is an example of sending tokens from the `tokenUser`/`tokenHolder` to another account:
+
+```typescript
+  const txn2 = await Mina.transaction(deployerAccount, () => {
+    let feePayerUpdate = AccountUpdate.fundNewAccount(deployerAccount, 1);
+    tokenUserInstance.sendMyTokens(UInt64.from(100), deployerAccount.toPublicKey());
+  });
+```
+
+The `fundNewAccount` call is included as the deployerAccount in this case doesn't have an existing token account.
+
+`tokenUserInstance.sendMyTokens(...)` is implemented as follows:
+
+
+```typescript
+  @method sendMyTokens(
+    amount: UInt64, 
+    destination: PublicKey
+  ) {
+    const tokenHolder = this.tokenHolder;
+    tokenHolder.transfer(destination, amount);
+    this.tokenContract.approveTransfer(tokenHolder.self);
+  }
+```
+
+Which calls the `transfer` contract on `tokenHolder`:
+
+```typescript
+  @method transfer(to: PublicKey, amount: UInt64) {
+    const toAccountUpdate = AccountUpdate.create(to, this.tokenId);
+    toAccountUpdate.balance.addInPlace(amount);
+    toAccountUpdate.body.mayUseToken =
+      AccountUpdate.MayUseToken.InheritFromParent;
+
+    this.balance.subInPlace(amount);
+
+    this.self.body.mayUseToken = AccountUpdate.MayUseToken.ParentsOwnToken;
+  }
+```
+
+And is approved by the token contract's `approveTransfer`:
+
+```typescript
+  @method approveTransfer(
+    transferUpdate: AccountUpdate
+  ) {
+    // TODO should anything else be asserted here for a "safe" deployment?
+    this.approve(transferUpdate);
+
+    // TODO should this also assert that all accountUpdates / balanceChanges are for the correct tokenId?
+    this.assertHasNoBalanceChange(transferUpdate);
+  }
+```
+
+This uses a helper `assertHasNoBalanceChange` which walks the accountUpdates being approved and ensures that the balance changes haven't created or destroyed any tokens.
+
+The account updates for this transcation looks as follows:
+
+<img src="/img/tutorial-11-txn2.png" width="95%" />
+
+This shows both the account update for the fee payment - and the main chain which does the token operations. You can see the call to `tokenUser` at top, followed by a call to `myToken`, that is doing the approving. This approves two accountUpdates - one of which is taking tokens away from the `tokenUser`, and has the `mayUseToken` field on - and the other of which is sending tokens to the deployer account.
+
+Because of the assertions in the `myToken` approval account, the balances must match out between these two account updates. Which it does, and so a valid `proof` can be constructed.
+
+As noted above, a token standard library is in the above which will abstract much of these details away too for standard use.
```

Comment:
```suggestion
This shows both the account update for the fee payment and the main chain that does the token operations. You can see the call to `tokenUser` at the top, followed by a call to `myToken` that is doing the approving. This approves two accountUpdates: one update takes tokens away from the `tokenUser` and has the `mayUseToken` field on, and the other update sends tokens to the deployer account.

Because of the assertions in the `myToken` approval account, the balances must match between these two account updates. Which it does, and so a valid `proof` can be constructed.

As noted earlier, a token standard library abstracts many of these details away for standard use.
```